src/udiskslinuxmanager.c

```diff
@@ -1224,6 +1230,27 @@ handle_resolve_device (UDisksManager         *object,
       if (devlabel != NULL)
           found = g_strcmp0 (udisks_block_get_id_label (blocks_p->data), devlabel) == 0;
 
+      if (partlabel != NULL || partuuid != NULL)
```

Comment:
These two checks should be separate -- you need to match both `partlabel` and `partuuid` when user specifies both (documentation for `ResolveDevice` allows specifying multiple keys and all must match.

Reply:
True, that OR between the tests doesn't conform to the _"devices matching all values"_ condition. 

Also, you should be pulling strings from the `org.freedesktop.UDisks2.Partition` interface of the block object as suggested in the D-Bus API docs addition. The udev `ID_PART_ENTRY_` properties are encoded strings (e.g. `ID_PART_ENTRY_NAME=EFI\x20System\x20Partition`) and are different from what is presented on the `.Partition` interface.

Reply:
FWIW, this code is borrowed from https://github.com/storaged-project/udisks/blob/udisks-2.9.2/src/udiskslinuxblock.c#L424. But from what I can tell `udisks_linux_block_matches_id` has a different use case as we don't have "matching all values" situation there.

Ok, I'll try to use `.Partition` interface. Maybe should be fixed in `udisks_linux_block_matches_id` as well?

Reply:
Indeed, thanks for bringing this up. I've tried to addess this in #888.